1.0.0 refactor

jgravois opened the pull request:
closes #74 

had to ditch the GIS day feature service because it doesn't look like its up anymore.

in the middle of refactoring i noticed that developers have to `new` up their 'Provider' objects because we don't do 
```js
EsriLeafletGeocoding.Controls.Geosearch.Providers.featureLayer = function(options){
  return new EsriLeafletGeocoding.Controls.Geosearch.Providers.FeatureLayer;
};
```
is that an intentional deviation?

patrickarlt commented on CHANGELOG.md:
```diff
@@ -1,5 +1,17 @@
+* Introduced support for dynamic suggestions from custom geocoding services. #65
+* Refactored code to account for changes introduced in Esri Leaflet `1.0.0`. #75
+* Fixed problem in `initialize` #63 (thanks @timwis!)
+Added support for the `dynamicLayers` option to `L.esri.DynamicMapLayer` https://github.com/Esri/esri-leaflet/issues/566
```
@jgravois could you remove this?

patrickarlt commented on package.json:
```diff
@@ -35,12 +35,13 @@
-    "leaflet": "^0.7.0",
-    "esri-leaflet": "^1.0.0-rc.5"
+    "leaflet": "0.7.3",
+    "esri-leaflet": "^1.0.0"
```
@jgravois can you check if the tests are passing locally by pointing this at `git@github.com:Esri/esri-leaflet.git` and then running `npm cache clean && npm install && npm test`? If they passing locally I'll merge this and the tests should then pass once we release Esri Leaflet 1.0.0.

jgravois commented:
i fixed the typo in the CHANGELOG and rebased.
i also followed your instructions and confirmed that tests were passing.  i did this yesterday by manually jamming the latest `esri-leaflet` content into the node_modules folder, but i'm glad to know how to do it the *real* way.

patrickarlt commented on package.json:
```diff
@@ -35,12 +35,13 @@
   "dependencies": {
-    "leaflet": "^0.7.0",
-    "esri-leaflet": "^1.0.0-rc.5"
+    "leaflet": "0.7.3",
```
Should probably change this to `^0.7.3` just in case there is a `0.7.4` release for Leaflet.

jgravois replied:
good call.  fixed.
